pkg/component/component_test.go

```diff
@@ -0,0 +1,92 @@
+package component
+
+import (
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/redhat-developer/odo/pkg/occlient"
+	"github.com/redhat-developer/odo/pkg/testingutil"
+	"k8s.io/apimachinery/pkg/runtime"
+	ktesting "k8s.io/client-go/testing"
+)
+
+func TestGetComponentPorts(t *testing.T) {
+	type args struct {
+		componentName   string
+		applicationName string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+		output  []string
+	}{
+		{
+			name: "Case 1: Invalid/Non-existant component name",
+			args: args{
+				componentName:   "r",
+				applicationName: "app",
+			},
+			wantErr: true,
+			output:  []string{},
+		},
+		{
+			name: "Case 2: Valid params with multiple containers each with multiple ports",
+			args: args{
+				componentName:   "python",
+				applicationName: "app",
+			},
+			output:  []string{"10080/TCP", "8080/TCP", "9090/UDP", "10090/UDP"},
+			wantErr: false,
+		},
+		{
+			name: "Case 3: Valid params with single container and single port",
+			args: args{
+				componentName:   "nodejs",
+				applicationName: "app",
+			},
+			output:  []string{"8080/TCP"},
+			wantErr: false,
+		},
+		{
+			name: "Case 4: Valid params with single container and multiple port",
+			args: args{
+				componentName:   "wildfly",
+				applicationName: "app",
+			},
+			output:  []string{"8090/TCP", "8080/TCP"},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			// Fake the client with the appropriate arguments
+			client, fakeClientSet := occlient.FakeNew()
+			fakeClientSet.AppsClientset.PrependReactor("list", "deploymentconfigs", func(action ktesting.Action) (bool, runtime.Object, error) {
+				//return true, testingutil.FakeDeploymentConfigs(tt.args.namespace, tt.args.componentName, tt.args.componentType, tt.args.applicationName, tt.args.containerPort), nil
```

Comment:
why aren't we using this one instead (generating the deploymentmentconfigs with passed in params)?

Reply:
only to avoid long list of parameters and multiple calls like if you see https://github.com/redhat-developer/odo/pull/610/files#diff-d89c4df0f7eae3a323f4958758b13435R28,

In some cases I have 2 containers per dc in some 1 and within a container, I have in some cases 2 ports exposed and in some, 1 which means the private functions in that file will also need to be exported. In fact the huge list of parameters vs long list of function calls made me feel would clutter the code here so felt it would be nice to keep it in the file where mock data is present :)

Reply:
okay. can you remove this line @anmolbabu since it's commented out?